scripts/ping360_socat.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python -u
+
+"""
+This manage socat sessions so that every Ping360 device has a screen session
+running socat to expose it to an UDP port.
+"""
+import time
+import subprocess
+import re
+
+def list_ping_devices():
+    """
+    List devices in dev/serial/ping/ping360*
+    """
+    try:
+        # Note: we could easily support Ping1D here.
+        output = subprocess.check_output("ls /dev/serial/ping/Ping360*", shell=True)
+        output = [line for line in output.decode().split("\n") if len(line) > 0]
+        return output
+    except subprocess.CalledProcessError as e:
+        print("Error listing devices at '/dev/serial/ping/ping360*':", e)
+        return []
+
+def device_has_screen(device):
+    """
+    Checks if 'device' has an associated screen session
+    returns true if so
+    """
+    screen_name = device.split("/")[-1].split("-t-")[0]
+    try:
+        output = subprocess.check_output(["sudo", "-Au", "pi", "screen", "-ls"], universal_newlines=True)
+        return output.decode()
+    except subprocess.CalledProcessError as e:
+        if e.returncode == 1:
+            output = e.output.decode() # screen v4.2.x always gives exit code of 1 for 'screen -ls'
+            if not output: # no stdout, just stderr (which we don't capture the contents)
+                print("Error reading screen sessions!")
+
+    regex_string = r"((?P<idNum>[0-9]+)\.(?P<idName>[^\t]+))(\t\(*(?P<time>[0-9][^\)]*)?\))?\t\((?P<state>[A-z]+)\)"
+    regex = re.compile(regex_string)
+    matches = regex.finditer(output)
+
+    sessions = [match.groupdict()['idName'].decode() for match in matches]
+    return screen_name in sessions
+
+def create_device_screen(device, port):
+    """
+    Creates a screen session for 'device' running socat to expose
+    at UDP port 'port'
+    """
+    try:
+        # Follow link to actual device
+        target_device = subprocess.check_output(' '.join(["readlink", "-f", "%s" % device]), shell=True)
+        # Strip newline from output
+        target_device = target_device.decode().split('\n')[0]
+    except Exception as e:
+        print("Error reading symlink: %s" % e)
+        return False
+    screen_name = device.split("/")[-1].split("-t-")[0]
+    command = "sudo -H -u pi screen -dm -S %s socat UDP-LISTEN:%s,fork,reuseaddr,ignoreeof FILE:%s,b2000000,raw,ignoreeof"
```

Comment:
-T and -t is missing, timeout is necessary thing since we are not using eof.

Reply:
it is working perfectly fine here. I'll run more tests.

Reply:
I guess this is only working because https://github.com/bluerobotics/ping-viewer/pull/592 isn't merged yet, right?

Reply:
Not entire related but yes.
Since EOF can be transmit in the stream as a normal byte, we should consider the usage of -T and -t.
From socat documentation:
```
-t<timeout>
When one channel has reached EOF, the write part of the other channel is shut down. Then, socat waits <timeout> [timeval] seconds before terminating. Default is 0.5 seconds. This timeout only applies to addresses where write and read part can be closed independently. When during the timeout interval the read part gives EOF, socat terminates without awaiting the timeout.
-T<timeout>
Total inactivity timeout: when socat is already in the transfer loop and nothing has happened for <timeout> [timeval] seconds (no data arrived, no interrupt occurred...) then it terminates. Useful with protocols like UDP that cannot transfer EOF.
```